docs/spec/reactors/consensus/proposer-selection.md

```diff
@@ -2,20 +2,186 @@
 
 This document specifies the Proposer Selection Procedure that is used in Tendermint to choose a round proposer.
 As Tendermint is “leader-based protocol”, the proposer selection is critical for its correct functioning.
-Let denote with `proposer_p(h,r)` a process returned by the Proposer Selection Procedure at the process p, at height h
-and round r. Then the Proposer Selection procedure should fulfill the following properties:
 
-`Agreement`: Given a validator set V, and two honest validators,
-p and q, for each height h, and each round r,
-proposer_p(h,r) = proposer_q(h,r)
+At a given block height, the proposer selection algorithm runs with the same validator set at each round .
+Between heights, an updated validator set may be specified by the application as part of the ABCIResponses' EndBlock. 
 
-`Liveness`: In every consecutive sequence of rounds of size K (K is system parameter), at least a
-single round has an honest proposer.
+## Requirements for Proposer Selection
 
-`Fairness`: The proposer selection is proportional to the validator voting power, i.e., a validator with more
-voting power is selected more frequently, proportional to its power. More precisely, given a set of processes
-with the total voting power N, during a sequence of rounds of size N, every process is proposer in a number of rounds
-equal to its voting power.
+This sections covers the requirements with Rx being mandatory and Ox optional requirements.
+The following requirements must be met by the Proposer Selection procedure:
+
+#### R1: Determinism
+Given a validator set `V`, and two honest validators `p` and `q`, for each height `h` and each round `r` the following must hold:
+
+  `proposer_p(h,r) = proposer_q(h,r)`
+
+where `proposer_p(h,r)` is the proposer returned by the Proposer Selection Procedure at process `p`, at height `h` and round `r`.
+
+#### R2: Liveliness
+In every consecutive sequence of rounds of size K (K is system parameter), at least a
+single round has an honest proposer (??)
+
+#### R3: Fairness
+In a stable network (no validator set changes) and for any sequence of K rounds with K > P, a validator v is elected as proposer with a frequency f proportional to its voting power VP(v) divided by the total voting power P:
+
+    f ~ VP(v) / P
```

Comment:
Question regarding the notation: Instead of `f`, should we call the frequency `f ᵥ ` , `f_v`, or `f(v)` to indicate it is the frequency of validator `v`?

Reply:
yes, fixed locally.